### PR TITLE
AADAdministrativeUnit enabling advanced query

### PR DESCRIPTION
**Pull request by ifinch**
#### Pull Request (PR) description
Implemented advanced query based on this
https://learn.microsoft.com/en-us/graph/aad-advanced-queries?tabs=http#administrative-unit-properties

#### This Pull Request (PR) fixes the following issues
None

### CODE REVIEW

**Comment by ifinch** [comment]
@NikCharlebois any idea why the unit test fails?

**Comment by NikCharlebois** [comment]
We will need to add an entry in the changelog.md file to describe the change.

**Comment by ifinch** [comment]
> Implemented advanced query based on this
> https://learn.microsoft.com/en-us/graph/aad-advanced-queries?tabs=http#administrative-unit-properties

Done

**Comment by FabienTschanz** [comment]
@ifinch The entry in the changelog should be in this branch and not in a separate pull request for better visibility. Or mention that those two pull requests are linked #4683 and #4700

**Comment by ifinch** [comment]
The #4683 (the change) and #4700 (changelog update) are linked .

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADAdministrativeUnit/MSFT_AADAdministrativeUnit.psm1 [hunk 1]
```diff
@@ -1010,7 +1010,52 @@ function Export-TargetResource
     {
         $Script:ExportMode = $true
         #region resource generator code
-        [array] $Script:exportedInstances = Get-MgBetaDirectoryAdministrativeUnit -Filter $Filter -All:$true -ErrorAction Stop
+        $ExportParameters = @{
+            Filter      = $Filter
+            All         = [switch]$true
+            Property    = $propertiesToRetrieve
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_AADAdministrativeUnit/MSFT_AADAdministrativeUnit.psm1:1016]: Where is this property coming from? Where is it defined?

**Review by NikCharlebois — CHANGES_REQUESTED** [review]

File: Modules/Microsoft365DSC/DSCResources/MSFT_AADAdministrativeUnit/MSFT_AADAdministrativeUnit.psm1 (hunk 1 above)
Comment (ifinch) [reply Modules/Microsoft365DSC/DSCResources/MSFT_AADAdministrativeUnit/MSFT_AADAdministrativeUnit.psm1]: it shouldn't be there, I've removed.

**Review by NikCharlebois — APPROVED** [review]

**Pull request merged**